### ISSUE TITLE
Questions about testing results

### ISSUE THREAD

**Issue by weleen** [OpenMotionLab/MotionGPT#49]
Thank you for your great job! I have tried to reproduce the results and encountered some issues.

Following instructions, I evaluate the provided checkpoint downloaded from [huggingface](https://huggingface.co/OpenMotionLab/MotionGPT-base).

I run the following commands:
```
python -m test --cfg configs/config_h3d_stage3.yaml --task t2m
python -m test --cfg configs/config_h3d_stage3.yaml --task m2t
```

The evaluation results are not consistent with the results reported in the paper. The attachments are the log and metrics.

t2m results:
<img width="563" alt="image" src="https://github.com/OpenMotionLab/MotionGPT/assets/7327734/9ec68798-1517-412e-ab41-1663d6bae754">
[log_2023-10-04-19-56-23_test.log](https://github.com/OpenMotionLab/MotionGPT/files/12804083/log_2023-10-04-19-56-23_test.log)
<img width="910" alt="image" src="https://github.com/OpenMotionLab/MotionGPT/assets/7327734/8b901bde-5786-4bd4-a6a9-85d2a0286e71">

Would you happen to have any idea about what's wrong with the configuration?

**Comment by weleen** [comment]
About m2t task, the testing process is stuck at the 4th replication since the SIGTERM signal.
<img width="1471" alt="image" src="https://github.com/OpenMotionLab/MotionGPT/assets/7327734/c0d07eea-63a8-43ab-9309-fb9da805ea9a">


Similar to t2m, the testing result is behind the results reported in the paper. Especially Bleu@4 and CIDEr, only around 6 and 7.
<img width="925" alt="image" src="https://github.com/OpenMotionLab/MotionGPT/assets/7327734/5c9f9c6f-3a93-4294-84ec-45483c624db5">

I would appreciate it if you have time to help fix my issue.😄

**Comment by LinghaoChan** [comment]
@weleen hi! Has this issue been resolved? We met the same issue.

**Comment by weleen** [comment]
> @weleen hi! Has this issue been resolved? We met the same issue.

I found some mistakes.

In m2t.py
https://github.com/OpenMotionLab/MotionGPT/blob/0499f16df4ddde44dfd72a7cbd7bd615af1b1a94/mGPT/metrics/m2t.py#L325-L329

In t2m.py
https://github.com/OpenMotionLab/MotionGPT/blob/0499f16df4ddde44dfd72a7cbd7bd615af1b1a94/mGPT/metrics/t2m.py#L251-L254

m_lens are divided two times.

**Comment by Spark001** [comment]
same issue

**Comment by GuangtaoLyu** [comment]
> @weleen hi! Has this issue been resolved? We met the same issue.

hi, me too.

**Comment by shin-wn** [comment]
me too.

**Comment by feifeifeiliu** [comment]
> > [@weleen](https://github.com/weleen) hi! Has this issue been resolved? We met the same issue.
> 
> [@LinghaoChan](https://github.com/LinghaoChan) I think there are some mistakes in get_motion_embeddings.
> 
> In m2t.py
> 
> [MotionGPT/mGPT/metrics/m2t.py](https://github.com/OpenMotionLab/MotionGPT/blob/0499f16df4ddde44dfd72a7cbd7bd615af1b1a94/mGPT/metrics/m2t.py#L325-L329)
> 
> Lines 325 to 329 in [0499f16](/OpenMotionLab/MotionGPT/commit/0499f16df4ddde44dfd72a7cbd7bd615af1b1a94)
> 
>  m_lens = torch.div(m_lens, 
>                     self.cfg.DATASET.HUMANML3D.UNIT_LEN, 
>                     rounding_mode="floor") 
>  ref_mov = self.t2m_moveencoder(feats_ref[..., :-4]).detach() 
>  m_lens = m_lens // self.unit_length 
> In t2m.py
> 
> [MotionGPT/mGPT/metrics/t2m.py](https://github.com/OpenMotionLab/MotionGPT/blob/0499f16df4ddde44dfd72a7cbd7bd615af1b1a94/mGPT/metrics/t2m.py#L251-L254)
> 
> Lines 251 to 254 in [0499f16](/OpenMotionLab/MotionGPT/commit/0499f16df4ddde44dfd72a7cbd7bd615af1b1a94)
> 
>  m_lens = torch.div(m_lens, 
>                     self.cfg.DATASET.HUMANML3D.UNIT_LEN, 
>                     rounding_mode="floor") 
>  m_lens = m_lens // self.cfg.DATASET.HUMANML3D.UNIT_LEN 
> m_lens are divided two times.
> 
> However, even I fix these errors, the results are still different. Have you solved this issus?

It looks like the variable 'm_lens' is not actually used:

```python
def forward(self, inputs, m_lens):

        num_samples = inputs.shape[0]

        input_embs = self.input_emb(inputs)
        hidden = self.hidden.repeat(1, num_samples, 1)

        cap_lens = m_lens.data.tolist()
        
        # emb = pack_padded_sequence(input=input_embs, lengths=cap_lens, batch_first=True)
        emb = input_embs

        gru_seq, gru_last = self.gru(emb, hidden)

        gru_last = torch.cat([gru_last[0], gru_last[1]], dim=-1)

        return self.output_net(gru_last)
```

**Comment by Lyman-Smoker** [comment]
I am also facing the same issue. 

Until now, I have tried the following modifications:

- I commented out the redundant division mentioned by @weleen ;
- I tried to  use ```emb = pack_padded_sequence(input=input_embs, lengths=cap_lens, batch_first=True)```  to use `m_lens`, according to the comments by @feifeifeiliu .

By doing so, the T2M results of the provided checkpoint (from [huggingface](https://huggingface.co/OpenMotionLab/MotionGPT-base)) are as follows:

![Image](https://github.com/user-attachments/assets/da6ba305-760a-4282-9092-01ffba3f9db0)


Compared with results posted by @weleen, after my modification Matching_score and R_precision get improved, but results on other metrics get even worse.

**Most importantly, after doing this, there is still a clear gap between these results and those posted in the paper, even if comparing with `MotionGPT(Pre-trained)`.**

Really hope the authors or someone who has successfully reproduced this work could provide some hints on this issue🙏
